File: 2.Arrays - CyclicRotation/answer.py
# you can write to stdout for debugging purposes, e.g.
# print("this is a debug message")

def solution(A, K):
    # write your code in Python 3.6
    ans = [0] * len(A)
    if(len(A) == 0):
        return A
    if(K > len(A)):
        K = K % len(A)
    for i in range(K, len(A)):
        ans[i] = A[i-K]
    for i in range(0, K):
        ans[i] = A[i-K]
    return ans

File: 6.Sorting - MaxProductOfThree/answer.js
// you can write to stdout for debugging purposes, e.g.
// console.log('this is a debug message');

function solution(A) {
    // write your code in JavaScript (Node.js 8.9.4)
    A = A.sort( function(a,b) { return a - b; } )
    let arrSize = A.length
    
    /* Obviously try to multiply from the array in highest order applicalble to all negative and all positive*/
    let answer = A[arrSize - 1] * A[arrSize - 2] * A[arrSize - 3];
    
    /* If the lowermost two negative multiply a positive is bigger */
    if(A[0] * A[1] * A[arrSize - 1] > answer)
        answer = A[0] * A[1] * A[arrSize - 1]
    
    
    return answer
}
